lib/svg4everybody.js

```diff
@@ -0,0 +1,114 @@
+/*! svg4everybody v1.0.0 | github.com/jonathantneal/svg4everybody */
+
+function embed(svg, g) {
+	if (g) {
+		var viewBox = !svg.getAttribute('viewBox') && g.getAttribute('viewBox');
+		var fragment = document.createDocumentFragment();
+		var clone = g.cloneNode(true);
+
+		if (viewBox) {
+			svg.setAttribute('viewBox', viewBox);
+		}
+
+		while (clone.childNodes.length) {
+			fragment.appendChild(clone.childNodes[0]);
+		}
+
+		svg.appendChild(fragment);
+	}
+}
+
+function loadreadystatechange(xhr) {
+	xhr.onreadystatechange = function () {
+		if (xhr.readyState === 4) {
+			var x = document.createElement('x');
+
+			x.innerHTML = xhr.responseText;
+
+			xhr.s.splice(0).map(function (array) {
+				embed(array[0], x.querySelector('#' + array[1].replace(/(\W)/g, '\\$1')));
+			});
+		}
+	};
+
+	xhr.onreadystatechange();
+}
+
+function svg4everybody(opts) {
+	opts = typeof opts === 'object' ? opts : {};
```

Comment:
I would say that's of the implementor to find out. Usually instead of duck-typing you should throw an error. Such as a short-intended library should not correct the developer's mistake. JS does not have strict typing but conventions are enough in this case IMO.